scrunch/datasets.py

```diff
@@ -680,7 +680,51 @@ def __readonly__(self, *args, **kwargs):
     del __readonly__
 
 
-class Dataset(object):
+class DatasetVariablesMixin(collections.Mapping):
+    """
+    Handles dataset variable iteration
+    """
+
+    def __init__(self, ds_resource):
+        self.ds_resource = ds_resource
+        self._vars = self.ds_resource.variables.index.items()
+
+    def __getitem__(self, item):
+        # Check if the attribute corresponds to a variable alias
+        variable = self.ds_resource.variables.by('alias').get(item)
```

Comment:
Following the comment above, we should have this cached so we don't do a GET to the variables catalog (which can be huge) each time we want to read a variable. Else we're fetching the same response over and over